### PR TITLE
Upgrade python & debian to the latest released version

### PR DESCRIPTION
**Pull request by jgadling**
### Description of proposed changes

Upgrade Python and Debian to the latest released version.

Python 3.7 is over four years old at this point. Our team has a variety of python scripts that we run alongside Nextstrain in Docker, and we'd like to be able to take advantage of the improved typing capabilities of Python 3.9+. In addition it upgrades to the latest debian LTS

### Related issue(s)

This supercedes #66 

### Testing

- [ ] Checks pass

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # image with tools only needed during the image build.
 
 # First build the temporary image.
-FROM python:3.7-slim-buster AS builder
+FROM python:3.10.8-slim-buster AS builder
```
Comment (tsibley) [inline Dockerfile:8]: It'd be safe to make this just `3.10` since we didn't pin patch releases previously.

**Comment by corneliusroemer** [comment]
Oh just saw this now and approved actions to run, thanks!

**Comment by tsibley** [comment]
I'll look at this today, along with some of the other lingering PRs in this repo.

**Comment by tsibley** [comment]
I addressed the Debian upgrade separately (#66) and reworked the changes here as #102. Will merge that when it passes CI.

**Pull request closed without merging**